src/components/RefinementList/RefinementList.tsx

```diff
@@ -310,7 +325,9 @@ class RefinementList<
     const facetValues = this.props.facetValues &&
       this.props.facetValues.length > 0 && (
         <ul className={this.props.cssClasses.list}>
-          {this.props.facetValues.map(this._generateFacetItem, this)}
+          {(this.props.facetValues as Array<
+            TRefinementListItem | HierarchicalMenuItem
+          >).map(this._generateFacetItem, this)}
```

Comment:
There is an issue in TypeScript, when you try to iterate on an array with an union type you get the error "This expression is not callable." https://github.com/microsoft/TypeScript/issues/36390

You have to manually cast it to avoid the error (https://github.com/microsoft/TypeScript/issues/36390#issuecomment-641718624)

According to this https://github.com/microsoft/TypeScript/commit/b217f22e798c781f55d17da72ed099a9dee5c650 it should be fixed in v4.2.3 but we are using the v4.1.5 in the project.

Reply:
```suggestion
          {
            // @ts-expect-error https://github.com/microsoft/TypeScript/commit/b217f22e798c781f55d17da72ed099a9dee5c650
            this.props.facetValues.map(this._generateFacetItem, this)
          }
```

something like this would fit best I think

Reply:
or updating in a parallel PR already now